plugins/modules/dns_auth_zone.py

```diff
@@ -2037,11 +2037,11 @@ def update(self):
 
         update_body = self.payload
 
-        if update_body.fqdn and update_body.fqdn != self.existing.fqdn:
+        if update_body.fqdn and update_body.fqdn == self.existing.fqdn:
```

Comment:
the condition is not quite right. If update_body.fqdn is None, then it will still go to the else part. That shouldn't happen since both the fields are marked as required, and so you can remove the first part of the condition. 

Also, you should change this to a function if you can, e.g.

```
    def validate_readonly_on_update(self, update_body, fields):
        for field in fields:
            if getattr(update_body, field) != getattr(self.existing, field):
                self.fail_json(msg=f"{field} cannot be updated")
            setattr(update_body, field, None)
```

Reply:
Fixed